app/controllers/admin/pages/templates_controller.rb

```diff
@@ -53,7 +54,8 @@ def delete
     flash[ :notice ] = t( 'template_deleted' ) if template.destroy
     redirect_to admin_pages_templates_path
   rescue ActiveRecord::NotNullViolation, ActiveRecord::RecordNotFound
```

Comment:
Consider removing the duplicated `rescue ...` lines throughout the `delete` controller actions by passing a block, e.g.

```rb
def delete
  handle_delete_exceptions(t('template_delete_failed'), admin_pages_templates_path) do
    template = PageTemplate.find( params[:id] )
    authorise template
    flash[ :notice ] = t( 'template_deleted' ) if template.destroy
    redirect_to admin_pages_templates_path
  end
end
```

```rb
def handle_delete_exceptions(alert_message, redirect_path)
  yield
rescue ActiveRecord::NotNullViolation, ActiveRecord::RecordNotFound
  skip_authorization
  redirect_to redirect_path, alert: alert_message
end
```